File: src/main/scala/com/carmanconsulting/akka/scala/ParameterizedHelloAkka.scala
package com.carmanconsulting.akka.scala

import akka.actor.Actor

class ParameterizedHelloAkka(format: String) extends Actor {

  def this() = {
    this("Hello, %s!")
  }

  override def receive: Receive = {
    case name: String =>
      val message: String = String.format(format, name)
      sender ! message
  }
}

File: src/main/java/com/carmanconsulting/akka/java/ParameterizedHelloAkka.java
package com.carmanconsulting.akka.java;

import akka.actor.UntypedActor;

public class ParameterizedHelloAkka extends UntypedActor {
//----------------------------------------------------------------------------------------------------------------------
// Fields
//----------------------------------------------------------------------------------------------------------------------

    public static final String DEFAULT_FORMAT = "Hello, %s!";

    private final String format;

//----------------------------------------------------------------------------------------------------------------------
// Constructors
//----------------------------------------------------------------------------------------------------------------------

    public ParameterizedHelloAkka() {
        this(DEFAULT_FORMAT);
    }

    public ParameterizedHelloAkka(String format) {
        this.format = format;
    }

//----------------------------------------------------------------------------------------------------------------------
// Canonical Methods
//----------------------------------------------------------------------------------------------------------------------

    @Override
    public void onReceive(Object message) {
        final String response = String.format(format, message);
        getSender().tell(response, getSelf());
    }
}
